tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -1007,7 +1007,7 @@ def i(...):
       return ParallelMapDataset(self, map_func, num_parallel_calls)
 
   def flat_map(self, map_func):
-    """Maps `map_func` across this dataset and flattens the result.
+    """Maps `map_func` across this dataset and flattens the result. Will produce identical results to 'tf.data.Dataset.interleave'
```

Comment:
I think this change might confuse readers. The output of `Dataset.flat_map()` is only identical to `Dataset.interleave()` when `cycle_length=1` (which is an unusual configuration, since it doesn't actually interleave any inputs!). I'm not sure it's worth introducing that concept to readers here.

Additional nits:
* By PEP 257, the first line of a docstring should be a one-line summary; and by the style guide, it's subject to the 80-character limit, so we can't add much more content at this exact location.
* We use an extended form of Markdown in docstrings, and symbol names should be enclosed in backticks ("`") to create hyperlinks for them.

Reply:
Alright. 
I created a quick example and replaced the "identical" with "similar".

Greets